linkerd/core/src/main/scala/io/buoyant/linkerd/Server.scala

```diff
@@ -30,12 +31,16 @@ trait Server {
   def port: Int
 
   def addr: InetSocketAddress = new InetSocketAddress(ip, port)
+
+  def announce: Seq[AnnouncerName]
 }
 
 object Server {
 
   case class RouterLabel(label: String)
 
+  case class AnnouncerName(service: String, version: Option[String])
```

Comment:
This feels a bit awkward to me. I think this should be a Path -- there's more information that needs to be encoded in some registry schemes (e.g. zone, environment, role), etc.  I don't think we should try to force this into a 2-tuple, and instead should provide a mechanism for these schemes to encode announcer-specific coordinates.